src/Microsoft.Data/StringColumn.cs

```diff
@@ -14,7 +14,7 @@ namespace Microsoft.Data
     /// A mutable column to hold strings
     /// </summary>
     /// <remarks> Is NOT Arrow compatible </remarks>
-    public partial class StringColumn : BaseColumn, IEnumerable<string>
+    public partial class StringColumn : BaseColumn, IEnumerable<List<string>>
```

Comment:
Same alternative here. We could leave the interface as it is and make a new API that returned a `List<string>`. We'd have to deal with keeping the `NullCount` in sync, but that is easily solved